modules/nixpkgs.nix

```diff
@@ -0,0 +1,137 @@
+# Licensed under GNU Lesser General Public License v3 or later, see COPYING.
+# Copyright (c) 2019 Alexander Sosedkin and other contributors, see AUTHORS.
+
+# Adapted from nixpkgs/home-manager.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  isConfig = x:
+    builtins.isAttrs x || builtins.isFunction x;
+
+  optCall = f: x:
+    if builtins.isFunction f
+    then f x
+    else f;
+
+  mergeConfig = lhs_: rhs_:
+    let
+      lhs = optCall lhs_ { inherit pkgs; };
+      rhs = optCall rhs_ { inherit pkgs; };
+    in
+    lhs // rhs //
+    optionalAttrs (lhs ? packageOverrides) {
+      packageOverrides = pkgs:
+        optCall lhs.packageOverrides pkgs //
+        optCall (attrByPath ["packageOverrides"] ({}) rhs) pkgs;
+    } //
+    optionalAttrs (lhs ? perlPackageOverrides) {
+      perlPackageOverrides = pkgs:
+        optCall lhs.perlPackageOverrides pkgs //
+        optCall (attrByPath ["perlPackageOverrides"] ({}) rhs) pkgs;
+    };
+
+  configType = mkOptionType {
+    name = "nixpkgs-config";
+    description = "nixpkgs config";
+    check = x:
+      let traceXIfNot = c:
+            if c x then true
+            else lib.traceSeqN 1 x false;
+      in traceXIfNot isConfig;
+    merge = args: fold (def: mergeConfig def.value) {};
+  };
+
+  overlayType = mkOptionType {
+    name = "nixpkgs-overlay";
+    description = "nixpkgs overlay";
+    check = builtins.isFunction;
+    merge = lib.mergeOneOption;
+  };
+
+  _pkgs = import <nixpkgs> (
+    filterAttrs (n: v: v != null) config.nixpkgs
+  );
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    nixpkgs = {
+      config = mkOption {
+        default = null;
+        example = { allowBroken = true; };
+        type = types.nullOr configType;
+        description = ''
+          The configuration of the Nix Packages collection. (For
+          details, see the Nixpkgs documentation.) It allows you to set
+          package configuration options.
+
+          </para><para>
+
+          If <literal>null</literal>, then configuration is taken from
+          the fallback location, for example,
+          <filename>~/.config/nixpkgs/config.nix</filename>.
+
+          </para><para>
+
+          Note, this option will not apply outside your Nix-on-Droid
+          configuration like when installing manually through
```

Comment:
And it won't apply to home-manager configuration either. Should it? If not, should we probably state that too and provide a snippet to pass it through to it?

Same goes for overlays.

Reply:
Like in nixpkgs there is also no such comment, but I can add a simple note